src/basic/compress.c

```diff
@@ -801,15 +853,19 @@ int decompress_stream_xz(int fdf, int fdt, uint64_t max_bytes) {
 int decompress_stream_lz4(int in, int out, uint64_t max_bytes) {
 #if HAVE_LZ4
         size_t c;
-        _cleanup_(LZ4F_freeDecompressionContextp) LZ4F_decompressionContext_t ctx = NULL;
+        _cleanup_(sym_LZ4F_freeDecompressionContextp) LZ4F_decompressionContext_t ctx = NULL;
         _cleanup_free_ char *buf = NULL;
         char *src;
         struct stat st;
         int r = 0;
```

Comment:
you can drop the zero init here now given you override it immediately anyway.

I think it would make sense to readd it though right before the `cleanup:` label at the end of the function, just to make sure we return 0 and not something else

Reply:
same thing applies to the compression function above

Reply:
you mean setting it to 0 in case we still hold the `1` from `dlopen_lz4()` or some size from `loop_write()`?

Also, where are we returning nonzero in `compress_stream_lz4()`?

Reply:
> you mean setting it to 0 in case we still hold the 1 from dlopen_lz4() or some size from loop_write()?

yes

Reply:
> Also, where are we returning nonzero in compress_stream_lz4()?

I have no idea what I was referring to. maybe ignore this comment ;-)